stack.go

```diff
@@ -6,47 +6,80 @@ import (
 	"strings"
 )
 
+// Stack is an array of stack frames stored in a human readable format.
+type Stack []StackFrame
+
+// contains checks if a stack trace contains a specific frame.
+func (s *Stack) contains(frame StackFrame) bool {
+	for _, f := range *s {
+		if f == frame {
+			return true
+		}
+	}
+	return false
+}
+
+// insertFrames inserts a wrap error frame into the correct place of the root error stack trace.
+func (s *Stack) insertFrame(wFrames []StackFrame) {
+	if len(wFrames) == 0 || s.contains(wFrames[0]) {
```

Comment:
why are we only checking for the first frame in wFrames and returning if it exists in the stack. Is it true that all the remaining frames in wFrames will be the same if wFrames[0] is in the stack?

Reply:
Yeah I'm pretty sure it is true. If the wrap frame is truly part of the caller stack, then the runtime.Callers output for wrapError should always be a subset of the runtime.Callers output for rootError (if that makes sense). This method assumes that you're trying to insert a frame that is part of the caller stack. If you provide something that isn't part of the caller stack, then it won't add anything to the rootError stack. I'm open to rethinking this if you see issues with it though.

Reply:
I also verified that it's true while writing this method, but I could've missed some edge case. Can you think of any example where what I said above isn't true?

Reply:
I see, what about the case when someone double/triple wraps an error like eris.Wrap(eris.Wrap(eris.Wrap(...)...)...)? Judging by the logic, each wrap will get its own stack and one frame from each will be inserted into the root error. So there will never be a case when wrap error has two distinct frames that are not part of the roorErroor?

Reply:
yeah, that's correct as far as i can tell. basically it seems like wFrames[0] is always the frame that won't be in the stack and wFrames[1] should always be in the stack. i included some examples that show how it works when wrap is called multiple times (as well as when errors are just returned without wrapping).